tinker/faculty_bios/__init__.py

```diff
@@ -215,3 +217,104 @@ def edit_all(self):
         xml_url = app.config['FACULTY_BIOS_XML_URL']
         self.base.edit_all(type_to_find, xml_url)
         return 'success'
+
+    @route("/faculty-bio-csv")
+    def get_faculty_bio_csv(self):
+
+        # Checks permissions to view/access route
+        if 'Administrators' not in session['groups'] or 'Tinker Faculty Bios - Admin' not in session['groups']:
+            abort(403)
+
+        # Traverses the xml file
+        info_form = self.base.traverse_xml(app.config['FACULTY_BIOS_XML_URL'], 'system-page', True, True)
+
+        # Opens the xml file and signifies that we will write to it
+        with open('faculty-info.csv', 'w') as csvfile:
+
+            filewriter = csv.writer(csvfile)
+
+            # # Adds column headers to the list
+            my_list = ['Faculty first name', 'Faculty last name', 'Faculty member\'s username', 'Location',
+                       'Highlight text']
+
+            max_jobs = 0
+            max_edu = 0
+            # Gets the maximum number of jobs and maximum number of educations out of everyone
+            for data in info_form:
+                if data['max-jobs'] > max_jobs:
+                    max_jobs = data['max-jobs']
+                if data['max-edu'] > max_edu:
+                    max_edu = data['max-edu']
+
+            # Creates max_jobs spaces for jobs and adds them to the list
+            for i in range(1, max_jobs + 1):
+                my_list.append('Job Title ' + str(i))
+                my_list.append('Job Title-School ' + str(i))
+                my_list.append('Job Title-Department ' + str(i))
+            #
+            my_list.extend(['Email', 'Started at Bethel in'])
+
+            # Creates max_edu spaces for educations and adds them to the list
+            for j in range(1, max_edu + 1):
+                my_list.append('Degree-Earned ' + str(j))
+                my_list.append('Degree-School ' + str(j))
+                my_list.append('Degree-Year ' + str(j))
+
+            # Adds the remaining column headers to the list
+            my_list.extend(['Biography', 'Courses Taught', 'Awards', 'Publications', 'Presentations',
```

Comment:
Maybe this is a bit nitpicky, but your code here might flow better if you put jobs/degrees(the dynamic ones) at the end. Then do all of the others at the beginning. Not a huge deal though, just a thought

Reply:
you can also do this in your csv method

Reply:
Did in csv method, in the __init__.py I did it the way displayed above to have a nice ordering that follows the faculty bio form on tinker.bethel.edu (and ordering matters when adding it to the list).

Reply:
Right. I was thinking it might be helpful to just throw it on the end for 2 reasons:
1) the code would be organized better/cleaner
2) When looking through the csv, it keeps all of the dynamic content together. Normally this isn't a problem, but when there are 20+ fields in a row that are empty for most people, and then they have email next, its kind of rough.

Reply:
Done